# HG changeset patch
# User Rik <rdrider0-list@yahoo.com>
# Date 1244052771 25200
#      Wed Jun 03 11:12:51 2009 -0700
# Branch release-3-2-x
# Node ID 818a05f4378edf09c05e2fc94c31eb91580e2e11
# Parent  0e60041ef2e523d7b5e87189dcd391fa31f8c670
Correct bugs in plot images produced for documentation

Initialized random number generator so hist plot is reproducible.
Fixed errorbar plot where errorbars were not centered on data.
Corrected typo in legend of interpft.
2009-06-03  Rik  <rdrider0-list@yahoo.com>

	* interpreter/plotimages.m: Correct errorbar plot for docs.  Make hist plot
		reprodroducible for docs.
			* interpreter/interpimages.m: Correct typo in legend of interpft plot for
				docs

diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,8 +1,15 @@
+2009-06-03  Rik  <rdrider0-list@yahoo.com>
+
+	* interpreter/plotimages.m: Correct errorbar plot for docs.  Make hist plot
+	reprodroducible for docs.
+	* interpreter/interpimages.m: Correct typo in legend of interpft plot for
+	docs 
+
 	2009-05-25  Jaroslav Hajek  <highegg@gmail.com>
 
 	Version 3.2.0 released.
 
 2009-06-03  Jaroslav Hajek  <highegg@gmail.com>
 
 	* Makefile.in: Omit conf.texi in DISTFILES.
 
diff --git a/doc/interpreter/interpimages.m b/doc/interpreter/interpimages.m
--- a/doc/interpreter/interpimages.m
+++ b/doc/interpreter/interpimages.m
@@ -26,17 +26,17 @@ function interpimages (nm, typ)
   elseif (strcmp (nm, "interpft"))
     t = 0 : 0.3 : pi; dt = t(2)-t(1);
     n = length (t); k = 100;
     ti = t(1) + [0 : k-1]*dt*n/k;
     y = sin (4*t + 0.3) .* cos (3*t - 0.1);
     yp = sin (4*ti + 0.3) .* cos (3*ti - 0.1);
     plot (ti, yp, 'g', ti, interp1(t, y, ti, 'spline'), 'b', ...
 	  ti, interpft (y, k), 'c', t, y, 'r+');
-    legend ('sin(4t+0.3)cos(3t-0.1','spline','interpft','data');
+    legend ('sin(4t+0.3)cos(3t-0.1)','spline','interpft','data');
     print (cstrcat (nm, ".", typ), cstrcat ("-d", typ))
   elseif (strcmp (nm, "interpn"))
     x = y = z = -1:1;
     f = @(x,y,z) x.^2 - y - z.^2;
     [xx, yy, zz] = meshgrid (x, y, z);
     v = f (xx,yy,zz);
     xi = yi = zi = -1:0.1:1;
     [xxi, yyi, zzi] = ndgrid (xi, yi, zi);
diff --git a/doc/interpreter/plotimages.m b/doc/interpreter/plotimages.m
--- a/doc/interpreter/plotimages.m
+++ b/doc/interpreter/plotimages.m
@@ -25,24 +25,27 @@ function plotimages (nm, typ)
   endif
   if (strcmp(typ , "txt"))
     image_as_txt(nm);
   elseif (strcmp (nm, "plot"))
     x = -10:0.1:10;
     plot (x, sin (x));
     print (cstrcat (nm, ".", typ), cstrcat ("-d", typ))    
   elseif (strcmp (nm, "hist"))
+	 rand ("state", 2);
     hist (randn (10000, 1), 30);
     print (cstrcat (nm, ".", typ), cstrcat ("-d", typ))    
   elseif (strcmp (nm, "errorbar"))
+	 rand ("state", 2);
     x = 0:0.1:10;
     y = sin (x);
-    yp =  0.1 .* randn (size (x));
-    ym = -0.1 .* randn (size (x));
-    errorbar (x, sin (x), ym, yp);
+    yl = 0.1 .* rand (size (x));
+    yu = 0.1 .* rand (size (x));
+    errorbar (x, sin (x), yl, yu);
+	 axis ([0, 10, -1.1, 1.1]);
     print (cstrcat (nm, ".", typ), cstrcat ("-d", typ))    
   elseif (strcmp (nm, "polar"))
     polar (0:0.1:10*pi, 0:0.1:10*pi);
     print (cstrcat (nm, ".", typ), cstrcat ("-d", typ))    
   elseif (strcmp (nm, "mesh"))
     tx = ty = linspace (-8, 8, 41)';
     [xx, yy] = meshgrid (tx, ty);
     r = sqrt (xx .^ 2 + yy .^ 2) + eps;
